id/news/_posts/2019-12-12-separation-of-positional-and-keyword-arguments-in-ruby-3-0.md

```diff
@@ -0,0 +1,473 @@
+untuk melewatkan sebuah objek *Hash* ketimbang *keyword* secara langsung.
+
+{% highlight ruby %}
+# Method ini menerima satu positional argument dan sebuah keyword rest argument
+def bar(h, **kwargs)
+  p h
+end
+
+# Pemanggilan ini hanya melewatkan sebuah keyword argument tanpa positional argument
+# Pada Ruby 2.7: keyword diubah ke sebuah positional Hash argument
+# Pada Ruby 3.0: Pemanggilan ini menyebabkan ArgumentError
+bar(k: 42)
+  # => demo2.rb:9: warning: Passing the keyword argument as the last hash parameter is deprecated
+  #    demo2.rb:2: warning: The called method `bar' is defined here
+  #    {:k=>42}
+
+# Jika Anda ingin tetap menggunakan perilaku sebelumnya pada Ruby 3.0, gunakan
+# tanda kurung kurawal untuk mengubahnya menjadi sebuah explicit Hash
+bar({ k: 42 }) # => {:k=>42}
+{% endhighlight %}
+
+## Apa yang sudah usang?
+{: #what-is-deprecated}
+
+Pada Ruby 2, *keyword argument* dapat dianggap sebagai *positional Hash argument*
+terakhir dan sebuah *positional Hash argument* terakhir dapat dianggap sebagai
+*keyword argument*.
+
+Karena pengubahan otomatis tersebut, kadang-kadang hal ini sangat kompleks dan
+sulit seperti yang dideskripsikan pada bagian terakhir. Sehingga, sekarang
+perilaku tersebut sudah usang pada Ruby 2.7 dan akan dihilangkan pada Ruby 3.
+Dengan kata lain, *keyword argument* akan dipisahkan seluruhnya dari
+*positional*-nya pada Ruby 3. Ketika Anda ingin melewatkan *keyword argument*,
+Anda seharusnya selalu menggunakan `foo(k: expr)` atau `foo(**expr)`. Jika
+Anda ingin menerima *keyword argument*, pada dasarnya Anda harus selalu
+menggunakan `def foo(k: default)` atau `def foo(k:)` atau `def foo(**kwargs)`.
+
+Ingat bahwa Ruby 3.0 tidak membedakan perilaku ketika memanggil sebuah *method*
+yang tidak menerima *keyword argument* dengan *method* yang menerima
+*keyword argument*. Sebagai contoh, kasus berikut ini tidak akan usang dan
+akan tetap berjalan pada Ruby 3.0. *Keyword argument* tetap dianggap sebagai
+sebuah *positional Hash argument*.
+
+{% highlight ruby %}
+def foo(kwargs = {})
+  kwargs
+end
+
+foo(k: 1) #=> {:k=>1}
+{% endhighlight %}
+
+Hal ini karena gaya di atas sangat sering digunakan dan tidak ada ambiguitas
+saat bagaimana *argument* seharusnya diperlakukan. Melarang pengubahan ini
+akan menyebabkan *incompatibility* tambahan untuk manfaat yang sedikit.
+
+Namun demikian, gaya ini tidak direkomendasikan pada kode baru, kecuali anda
+sering melewatkan sebuah *Hash* sebagai sebuah *positional argument* dan juga
+menggunakan *keyword argument*. Jika tidak, gunakan *double splat*:
+
+{% highlight ruby %}
+def foo(**kwargs)
+  kwargs
+end
+
+foo(k: 1) #=> {:k=>1}
+{% endhighlight %}
+
+## Akankah kode saya tidak berjalan pada Ruby 2.7?
+{: #break-on-ruby-2-7 }
+
+Jawaban singkatnya adalah "mungkin tidak".
+
+Perubahan pada Ruby 2.7 didesain sebagai sebuah migrasi menuju 3.0. Pada
+dasarnya, Ruby 2.7 hanya memeringatkan perilaku yang akan berubah pada Ruby 3,
```

Comment:
I think it should be "memperingatkan" instead of "memeringatkan"

Reply:
Yeah, this one should be "memperingatkan". Updated @kuntoaji.

Reply:
If there is no other feedback, I will merge this @kuntoaji